model/src/funTest/kotlin/utils/PostgresProvenanceFileStorageFunTest.kt

```diff
@@ -35,7 +35,7 @@ import org.ossreviewtoolkit.model.VcsType
 import org.ossreviewtoolkit.model.config.FileArchiverConfiguration
 import org.ossreviewtoolkit.utils.test.PostgresListener
 
-private val SOURCE_ARTIFACT_PROVENANCE = ArtifactProvenance(sourceArtifact = RemoteArtifact("url", Hash.create("hash")))
+private val SOURCE_ARTIFACT_PROVENANCE = ArtifactProvenance(RemoteArtifact("https://example.com/", Hash.NONE))
```

Comment:
Would it make sense to add an alternative constructor which does not have these sanity checks, so that we can more easily choose values in tests?

Reply:
I was also a bit torn between the options. As the `Hash` class is also part of the serialized data / file model, IMO the `data class` itself indeed should not have such sanity checks. Instead, an interface that abstracts away the concrete data storage format should have these checks. As this is a reoccurring topic I've created [this epic](https://github.com/oss-review-toolkit/ort/issues/8914) for it.

But for now, I believe it's a good change to add some sanity and not make data *too* fake.